### PR TITLE
fix: convert Date automatically created from YAML into a string

### PR DESCRIPTION
**Pull request by dominykas**
- [ ] Ready for review
- [ ] Follows CONTRIBUTING rules
- [ ] Reviewed by @remy (Snyk internal team)

#### What does this PR do?

Fixes https://github.com/snyk/snyk/issues/120

#### How should this be manually tested?

See https://github.com/snyk/snyk/issues/120

#### Additional questions

I didn't add any tests for the notes part - there weren't any that deal with expired notes to start with. I suspect that code might be unreachable, because these deps should be filtered out during load time? I can revisit later, unless someone gets to it sooner...

### CODE REVIEW

**Comment by odinn1984** [comment]
@dominykas please squash the commits and use "fix:" in the commit message and in the PR title.

**Comment by dominykas** [comment]
done

**Comment by dominykas** [comment]
Hey, any update on this?

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 /node_modules/
 /coverage
 tmp
+.idea
```
Comment (adrukh) [inline .gitignore]: A bit nit-picky, but can you please remove this? We use global ignores for our own envs (like IDE of choice), leaving the project's ignore for files specifically relevant to the project.

Comment (dominykas) [reply .gitignore]: Done.

I avoid global ignores myself, because IIRC npm does not respect the global `.gitignore` during `npm publish`.

**Pull request merged**